src/providers/token-properties-provider.ts

```diff
@@ -11,14 +13,15 @@ import {
 } from './token-fee-fetcher';
 import {
   DEFAULT_ALLOWLIST,
-  ITokenValidatorProvider,
   TokenValidationResult,
 } from './token-validator-provider';
-import { BigNumber } from '@ethersproject/bignumber';
+
 
 export const DEFAULT_TOKEN_PROPERTIES_RESULT: TokenPropertiesResult = {
   tokenFeeResult: DEFAULT_TOKEN_FEE_RESULT,
 };
+export const POSITIVE_CACHE_ENTRY_TTL = 600; // 10 minutes in seconds
+export const NEGATIVE_CACHE_ENTRY_TTL = 10; // 10 seconds
```

Comment:
Just curioous why is the negative cache ttl so small?

Reply:
It's a heuristic based on the mobile UX feedback. When we mistakenly cache the FOT token with no fees, on the clients side, it shows the UI doesn't populate the FOT taxes. This seems like a pretty bad UX. By making it shorter, we can make the router refresh and return the FOT tax sooner, just in case if we cache the FOT token with no fees for some reason, e.g. bug, failed contract calls, etc.

Since the token properties provider constructor accepts both positive and negative TTLs, it's easy to adjust on the routing-api side, as we roll out and learn more UX.

Reply:
I think the concern is that the majority of tokens will be in the set of "negative" cache, and if we ttl every 10 seconds, then thats a lot of rpc calls.

Could we set this to 10 minutes too and see how bad it is, before we go with doing it 10 seconds from the get go?

Reply:
sure